### ISSUE TITLE
RandomTransplantation inside AugmentationSequential raises unless it is the first op: an earlier child promotes the mask to (B, 1, H, W)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4707]
## Summary

`RandomTransplantation` inside `AugmentationSequential` works **only when no other augmentation runs
before it**. Any earlier `_AugmentationBase` child — even one whose gate is closed — leaves the mask as
`(B, 1, H, W)`, and the transplant then refuses it, because it requires the image to have exactly one
more axis than the mask. The container's own documented mask layout `(B, 1, H, W)` is refused for the
same reason even when the transplant is the first op.

The class docstring says it is "intended to be used with `AugmentationSequential`", and a realistic
pipeline (flip or normalize, then transplant) is exactly the case that raises.

## Repro

```python
import torch
import kornia.augmentation as K

img = torch.rand(3, 2, 4, 6)
mask = torch.randint(0, 3, (3, 4, 6))
T = lambda: K.RandomTransplantation(p=1.0)

def run(name, *ops, m=mask):
    try:
        out = K.AugmentationSequential(*ops, data_keys=["image", "mask"])(img, m)
        print(name, "OK", [tuple(o.shape) for o in out])
    except Exception as e:
        print(name, "RAISED", type(e).__name__, str(e)[:95])

run("transplant alone            ", T())
run("HFlip(p=0) then transplant  ", K.RandomHorizontalFlip(p=0.0), T())
run("Normalize then transplant   ", K.Normalize(0.5, 0.5), T())
run("transplant then HFlip(p=0)  ", T(), K.RandomHorizontalFlip(p=0.0))
run("transplant alone, (B,1,H,W) ", T(), m=mask[:, None])
```

## Observed (main @ 5390ec29d, torch 2.14.0, CPU)

```
transplant alone             OK [(3, 2, 4, 6), (3, 4, 6)]
HFlip(p=0) then transplant   RAISED BaseError Every image input must have one additional dimension (channel dimension) than the segmentation mask, but got 4 for the input image and 4 for the segmentation mask.
Normalize then transplant    RAISED BaseError (same message)
transplant then HFlip(p=0)   OK [(3, 2, 4, 6), (3, 1, 4, 6)]
transplant alone, (B,1,H,W)  RAISED BaseError (same message)
```

The traceback ends in `RandomTransplantation.params_from_input`, called from
`kornia/augmentation/container/ops.py` (`AugmentationSequentialOps.transform`). The earlier child's
mask op has already promoted the mask to `(B, 1, H, W)`, so `_input.ndim == mask.ndim + 1` fails.
A preceding `nn.Identity` does not trigger it. `RandomTransplantation3D` behaves the same way with
`(B, 1, D, H, W)`.

Two smaller consequences of the same mismatch:

- the returned mask rank depends on the pipeline: `(B, H, W)` when the transplant is the only op,
  `(B, 1, H, W)` when anything follows it. The `AugmentationSequential` docstring says a `(B, H, W)`
  mask "is accepted and returned as `(B, 1, H, W)`", which is not what a transplant-only pipeline does.
- the check that fails is a rank check, so the message talks about a missing channel dimension on the
  *image*, which is not what is wrong.

## Not a regression

Identical on the `v0.7.3` and `v0.8.1` tags (same repro, `Exception: False not true. Every image input
must have one additional dimension ...`), so this has been the behaviour since the class was added.
The existing container test for the class, `test_repeating[AugmentationSequential]`, only ever builds
a pipeline with the transplant as its single op.

## Classification

A defect, not a convention: the documented intended use raises. The likely fix is on the
transplantation side — accept a mask carrying a singleton channel axis (`mask.ndim == image.ndim` with
`mask.shape[1] == 1`) and select through it — rather than in the container, since every other mask
consumer already lives with the promoted layout. Found while reviewing #4695 (batch 6e of #4407);
that PR pins the current behaviour as a wart and does not change it. Related container-scope items from
the same audit: #4692, #4693.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
